Sources/MongoSwift/MongoError.swift

```diff
@@ -481,4 +481,28 @@ internal func wrongIterTypeError(_ iter: DocumentIterator, expected type: BSONVa
     )
 }
 
+internal func isMaxTimeMSExpiredError(_ error: Error) -> Bool {
+    let maxTimeMSExpiredErrorCode = 50
+
+    if let commandError = error as? CommandError, commandError.code == maxTimeMSExpiredErrorCode {
+        return true
+    } else if let writeError = error as? WriteError {
+        if writeError.writeFailure?.code == maxTimeMSExpiredErrorCode ||
+            writeError.writeConcernFailure?.code == maxTimeMSExpiredErrorCode {
+            return true
+        }
+    } else if let bulkWriteError = error as? BulkWriteError {
+        if let writeFailures = bulkWriteError.writeFailures {
+            for writeFailure in writeFailures where writeFailure.code == maxTimeMSExpiredErrorCode {
```

Comment:
this could be written a bit more succinctly like
```swift
return writeFailures.contains { $0.code == maxTimeMSExpiredErrorCode }
```

Reply:
Doesn't this mean that you would return `false` if no `writeFailures` match the code, skipping the check below for `writeConcernFailure`?

Reply:
oh whoops, good point :)